pkg/code-generator/docgen/generator.go

```diff
@@ -147,22 +147,37 @@ func (d *DocsGen) FileHeader(filename string) string {
 		return ".. Code generated by solo-kit. DO NOT EDIT."
 	}
 	if d.DocsOptions.Output == options.Hugo {
-		name := filepath.Base(filename)
-		if strings.HasSuffix(name, d.protoSuffix()) {
-			name = name[:len(name)-len(d.protoSuffix())]
-		}
-		return fmt.Sprintf(`
+		return d.hugoFileHeader(filename)
+	}
+	return `<!-- Code generated by solo-kit. DO NOT EDIT. -->
+`
+}
+
+func (d *DocsGen) hugoFileHeader(filename string) string {
+	name := filepath.Base(filename)
+
+	var title string
+	if strings.HasSuffix(name, d.protoSuffix()) {
+		// Remove the "sk.md" extensions
+		name = name[:len(name)-len(d.protoSuffix())]
+
+		// Remove the ".proto" extension
+		extension := filepath.Ext(name)
+		title = strcase.ToCamel(name[0 : len(name)-len(extension)])
```

Comment:
If it's always the same ".proto.sk.md" extension, why not do this in one modification?
```suggestion
        protoExtension := ".proto"+d.protoSuffix()
	if strings.HasSuffix(name, protoExtension) {
		// Remove the "proto.sk.md" extension
		name = name[:len(name)-len(protoExtension)]

		title = strcase.ToCamel(name)
```